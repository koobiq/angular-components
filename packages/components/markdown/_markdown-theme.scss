@use 'sass:meta';
@use 'sass:map';

@use '../core/styles/theming/theming' as *;
@use '../core/styles/typography/typography-utils' as *;
@use '../core/styles/common/tokens' as *;


@mixin kbq-markdown-theme($theme) {
    $markdown: map.get($theme, components, markdown);

    .kbq-markdown {
        color: kbq-css-variable(foreground-contrast, map.get($theme, foreground, contrast));
        background: kbq-css-variable(background-bg, map.get($theme, background, bg));

        // h1, h2, h3, h4, h5, h6
        @for $i from 1 through 6 {
            .kbq-markdown__h#{$i} {
                color: kbq-css-variable(markdown-h#{$i}-color, map.get($markdown, h#{$i}-color));
            }
        }

        .kbq-markdown__p {
            color: kbq-css-variable(markdown-p-color, map.get($markdown, p-color));
        }

        .kbq-markdown__ul,
        .kbq-markdown__ol {
            color: kbq-css-variable(markdown-list-color, map.get($markdown, list-color));
        }

        .kbq-markdown__blockquote {
            color: kbq-css-variable(markdown-blockquote-text, map.get($markdown, blockquote-text));
            background: kbq-css-variable(markdown-blockquote-background, map.get($markdown, blockquote-background));
            border-color: kbq-css-variable(markdown-blockquote-border, map.get($markdown, blockquote-border));
            border-left-color: kbq-css-variable(markdown-blockquote-line, map.get($markdown, blockquote-line));
        }

        .kbq-markdown__pre,
        .kbq-markdown__p > .kbq-markdown__code {
            color: kbq-css-variable(markdown-code-text, map.get($markdown, code-text));
            background-color: kbq-css-variable(markdown-code-background, map.get($markdown, code-background));
            border-color: kbq-css-variable(markdown-code-border, map.get($markdown, code-border));
        }

        .kbq-markdown__a {
            color: kbq-css-variable(markdown-link-text, map.get($markdown, link-text));
            border-bottom-color: kbq-css-variable(markdown-link-border-bottom, map.get($markdown, link-border-bottom));

            &:visited {
                color: kbq-css-variable(markdown-link-state-visited-text, map.get($markdown, link-state-visited-text));
                border-bottom-color: kbq-css-variable(
                    markdown-link-state-visited-border-bottom, map.get($markdown, link-state-visited-border-bottom));
            }

            &:hover {
                color: kbq-css-variable(markdown-link-state-hover-text, map.get($markdown, link-state-hover-text));
                transition: color 0ms;
                border-bottom-color: kbq-css-variable(
                    markdown-link-state-hover-border-bottom, map.get($markdown, link-state-hover-border-bottom));
            }

            &:active {
                color: kbq-css-variable(markdown-link-state-active-text, map.get($markdown, link-state-active-text));
            }

            &:focus {
                outline:
                    kbq-css-variable(
                        markdown-link-state-focused-outline, map.get($markdown, link-state-focused-outline))
                    solid
                    kbq-css-variable(markdown-link-size-state-focused-outline-width);

                outline-offset: kbq-css-variable(markdown-link-size-state-focused-outline-offset);
            }
        }

        .kbq-markdown__img + em {
            color: kbq-css-variable(markdown-image-caption-text, map.get($markdown, image-caption-text));
        }

        .kbq-markdown__hr {
            border-bottom-color: kbq-css-variable(markdown-hr-color, map.get($markdown, hr-color));
        }

        .kbq-markdown__table > .kbq-markdown__thead {
            color: kbq-css-variable(markdown-table-header, map.get($markdown, table-header));
            border-bottom-color: kbq-css-variable(markdown-table-border, map.get($markdown, table-border));
        }

        .kbq-markdown__table > .kbq-markdown__tbody {
            color: kbq-css-variable(markdown-table-body, map.get($markdown, table-body));
        }
    }
}

@mixin kbq-markdown-typography($config) {
    .kbq-markdown {
        // h1, h2, h3, h4, h5, h6
        @for $i from 1 through 6 {
            .kbq-markdown__h#{$i} {
                @include kbq-typography-css-variables(markdown-h#{$i}, default);
            }
        }

        .kbq-markdown__p {
            @include kbq-typography-css-variables(markdown-p, default);
        }

        .kbq-markdown__ul,
        .kbq-markdown__ol {
            @include kbq-typography-css-variables(markdown-list, default);
        }

        .kbq-markdown__blockquote > .kbq-markdown__p {
            @include kbq-typography-css-variables(markdown-blockquote, default);
        }

        .kbq-markdown__code {
            @include kbq-typography-css-variables(markdown-code, default);
        }

        .kbq-markdown__a {
            @include kbq-typography-css-variables(markdown-link, default);
        }

        .kbq-markdown__img + em {
            @include kbq-typography-css-variables(markdown-image, caption);
        }

        .kbq-markdown__table > .kbq-markdown__thead {
            @include kbq-typography-css-variables(markdown-table, header);
        }

        .kbq-markdown__table > .kbq-markdown__tbody {
            @include kbq-typography-css-variables(markdown-table, body);
        }
    }
}
