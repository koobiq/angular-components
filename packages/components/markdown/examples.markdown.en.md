#### Headers combinations

<!-- example(markdown-headers-combinations) -->

#### Large article

<!-- example(markdown-article) -->
