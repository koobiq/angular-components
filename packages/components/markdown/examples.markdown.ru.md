#### Комбинация заголовков

<!-- example(markdown-headers-combinations) -->

#### Большая статья

<!-- example(markdown-article) -->
