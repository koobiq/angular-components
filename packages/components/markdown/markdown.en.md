`KbqMarkdown` - - a component that allows converting text written in _Markdown_ markup language into _HTML_.

<div class="kbq-callout kbq-callout_theme">
<div class="kbq-callout__header">Note</div>
<div class="kbq-callout__content kbq-docs-element-last-child-margin-bottom-0">

For the component to work, the [`marked@^15`](https://github.com/markedjs/marked/tree/v15.0.7) dependency is required:

```bash
npm install marked@^15
```

</div>
</div>

### Headers

<!-- example(markdown-headers) -->

### Paragraph

<!-- example(markdown-paragraph) -->

### Text emphasis

<!-- example(markdown-selection) -->

### Blockquote

<!-- example(markdown-quote) -->

### Lists

<!-- example(markdown-list) -->

### Inline code

<!-- example(markdown-code) -->

### Code block

<!-- example(markdown-code-block) -->

### Horizontal rule

<!-- example(markdown-divider) -->

### Link

<!-- example(markdown-link) -->

### Image

<!-- example(markdown-image) -->

### Table

<!-- example(markdown-table) -->

### Line break

<!-- example(markdown-line-break) -->
