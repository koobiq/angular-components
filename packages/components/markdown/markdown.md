`kbq-markdown` - компонент, который позволяет преобразовывать текст, написанный на языке разметки `Markdown`, в `HTML`.

**Обрати внимание!** Для работы компонента, необходимо наличие [`marked`](https://github.com/markedjs/marked) зависимости:

```bash
npm install marked@^9
```

### Заголовки

<!-- example(markdown-headers) -->

### Абзац

<!-- example(markdown-paragraph) -->

### Выделение текста

<!-- example(markdown-selection) -->

### Цитата

<!-- example(markdown-quote) -->

### Списки

<!-- example(markdown-list) -->

### Код

<!-- example(markdown-code) -->

### Блок кода

<!-- example(markdown-code-block) -->

### Горизонтальный разделитель

<!-- example(markdown-divider) -->

### Ссылка

<!-- example(markdown-link) -->

### Изображение

<!-- example(markdown-image) -->

### Таблица

<!-- example(markdown-table) -->

### Перенос строки

<!-- example(markdown-line-break) -->
