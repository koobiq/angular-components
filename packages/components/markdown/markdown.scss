@use 'sass:meta';
@use 'sass:map';

@use '../core/styles/tokens';
@use '../core/styles/common/tokens' as *;

$tokens: meta.module-variables(tokens) !default;

.kbq-markdown {
    display: block;

    .markdown-input {
        display: none;
    }

    .markdown-output {
        flex: 1;
    }

    // h1, h2, h3, h4, h5, h6
    @for $i from 1 through 6 {
        .kbq-markdown__h#{$i} {
            margin-top: kbq-css-variable(markdown-h#{$i}-size-margin-top);
            margin-bottom: kbq-css-variable(markdown-h#{$i}-size-margin-bottom);
        }
    }

    .kbq-markdown__a {
        display: inline-block;

        text-decoration: none;
        cursor: pointer;

        border-bottom-style: solid;
        border-bottom-width: 1px;
        transition: color ease-out 300ms;
    }

    .kbq-markdown__p {
        margin-top: kbq-css-variable(markdown-p-size-margin-top);
        margin-bottom: kbq-css-variable(markdown-p-size-margin-bottom);
    }

    .kbq-markdown__ul,
    .kbq-markdown__ol {
        margin-top: kbq-css-variable(markdown-list-size-margin-top);
        margin-bottom: kbq-css-variable(markdown-list-size-margin-bottom);

        > li {
            margin-bottom: kbq-css-variable(markdown-list-size-item-margin-bottom);
        }
    }

    .kbq-markdown__ul {
        position: relative;
        list-style: none;
        padding: kbq-css-variable(markdown-list-size-ul-padding);
    }

    .kbq-markdown__ul > li:before {
        content: 'â€”';
        display: block;
        position: absolute;
        width: 24px;
        text-align: left;
        left: 0;
    }

    .kbq-markdown__ol {
        list-style: none;
        padding: 0;
        counter-reset: li;

        > li {
            &:before {
                content: counter(li) '.';
                counter-increment: li;
                padding-right: kbq-css-variable(markdown-list-size-ol-number-padding-right);
            }

            > * {
                display: inline-block;
            }
        }
    }

    .kbq-markdown__p + .kbq-markdown__ul,
    .kbq-markdown__p + .kbq-markdown__ol {
        margin-top: kbq-css-variable(markdown-list-size-margin-top-after-paragraph);
    }

    .kbq-markdown__blockquote {
        margin-top: kbq-css-variable(markdown-blockquote-size-margin-top);
        margin-left: 0;
        margin-bottom: kbq-css-variable(markdown-blockquote-size-margin-bottom);
        margin-right: 0;

        border-left-width: kbq-css-variable(markdown-blockquote-size-line-width);

        border-left-style: solid;

        padding: kbq-css-variable(markdown-blockquote-size-padding);
    }

    .kbq-markdown__blockquote > .kbq-markdown__p {
        margin-top: 0;
        margin-bottom: 0;
    }

    .kbq-markdown__pre {
        position: relative;
        overflow-x: auto;

        margin-top: kbq-css-variable(markdown-code-size-multiline-margin-top);
        margin-bottom: kbq-css-variable(markdown-code-size-multiline-margin-bottom);

        padding: kbq-css-variable(markdown-code-size-multiline-padding);
    }

    .kbq-markdown__pre,
    .kbq-markdown__p > .kbq-markdown__code {
        border-radius: kbq-css-variable(markdown-code-size-border-radius);
        border-width: kbq-css-variable(markdown-code-size-border-width);
        border-style: solid;
    }

    .kbq-markdown__p > .kbq-markdown__code {
        padding: kbq-css-variable(markdown-code-size-inline-padding);
    }

    .kbq-markdown__img {
        max-width: 100%;
        margin-top: kbq-css-variable(markdown-image-size-margin-top);
        margin-bottom: kbq-css-variable(markdown-image-size-margin-bottom);

        + em {
            display: block;
            margin-top: kbq-css-variable(markdown-image-size-caption-margin-top);
            margin-bottom: kbq-css-variable(markdown-image-size-caption-margin-bottom);
        }
    }

    .kbq-markdown__hr {
        border-top: none;
        border-left: none;
        border-right: none;
        border-bottom-style: solid;
        border-bottom-width: kbq-css-variable(markdown-hr-size-width);
        margin: kbq-css-variable(markdown-hr-size-margin-vertical) 0;
    }

    .kbq-markdown__table {
        max-width: 100%;
        overflow-x: auto;
        display: table;
        border-collapse: collapse;
        margin-bottom: kbq-css-variable(markdown-table-size-margin-bottom);

        td {
            vertical-align: top;
        }

        th,
        td {
            padding: kbq-css-variable(markdown-table-size-padding);
            font-weight: 400;
        }
    }

    .kbq-markdown__table > .kbq-markdown__thead {
        border-bottom-style: solid;
        border-bottom-width: kbq-css-variable(markdown-table-size-border-width);

        text-align: left;

        th:first-child {
            padding-left: 0;
        }

        th:last-child {
            padding-right: 0;
        }
    }

    .kbq-markdown__table > .kbq-markdown__tbody {
        td:first-child {
            padding-left: 0;
        }

        td:last-child {
            padding-right: 0;
        }
    }
}

.docs-live-example.kbq-markdown {
    margin-bottom: 128px;
}
