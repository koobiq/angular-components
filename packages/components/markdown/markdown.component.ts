import {
    afterNextRender,
    ChangeDetectionStrategy,
    ChangeDetectorRef,
    Component,
    ElementRef,
    Input,
    ViewChild,
    ViewEncapsulation
} from '@angular/core';
import { KbqSafeHtmlPipe } from '@koobiq/components/core';
import { KbqMarkdownService } from './markdown.service';

/** Component which allows to convert `Markdown` into `HTML` */
@Component({
    standalone: true,
    imports: [KbqSafeHtmlPipe],
    selector: 'kbq-markdown',
    styleUrls: ['./markdown.scss', 'markdown-tokens.scss'],
    // no need format line with ng-content it's broke textContent for markdownService.parseToHtml()
    template: `
        <pre
            class="markdown-input"
            #contentWrapper
            ngPreserveWhitespaces
        ><ng-content /></pre>
        <div
            class="markdown-output"
            [innerHtml]="resultHtml | kbqSafeHtml"
        ></div>
    `,
    host: {
        class: 'kbq-markdown'
    },
    changeDetection: ChangeDetectionStrategy.OnPush,
    encapsulation: ViewEncapsulation.None
})
export class KbqMarkdown {
    @ViewChild('contentWrapper', { static: true }) private readonly contentWrapper: ElementRef;

    /** `Markdown` text. */
    @Input()
    get markdownText(): string | null {
        return this._markdownText;
    }

    set markdownText(value: string | null) {
        if (value && this.markdownText !== value) {
            this.resultHtml = this.markdownService.parseToHtml(value);
        }

        this._markdownText = value;
    }

    private _markdownText: string | null = null;

    protected resultHtml: string = '';

    constructor(
        private readonly markdownService: KbqMarkdownService,
        private readonly cdr: ChangeDetectorRef
    ) {
        afterNextRender(() => {
            if (!this.markdownText && this.contentWrapper?.nativeElement.textContent) {
                this.resultHtml = this.markdownService.parseToHtml(this.contentWrapper.nativeElement.textContent);
                this.cdr.detectChanges();
            }
        });
    }
}
