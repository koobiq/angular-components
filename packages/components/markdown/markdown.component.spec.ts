import { ChangeDetectionStrategy, Component, DebugElement, Type } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { KbqMarkdown, KbqMarkdownModule } from '@koobiq/components/markdown';

const createComponent = async <T>(component: Type<T>, providers: any[] = []): Promise<ComponentFixture<T>> => {
    await TestBed.configureTestingModule({ imports: [component], providers }).compileComponents();
    const fixture = TestBed.createComponent<T>(component);
    fixture.autoDetectChanges();
    return fixture;
};

const getMarkdownDebugElement = (debugElement: DebugElement): DebugElement => {
    return debugElement.query(By.directive(KbqMarkdown));
};

@Component({
    standalone: true,
    imports: [KbqMarkdownModule],
    selector: 'generate-html-from-markdown-string',
    template: `
        <kbq-markdown [markdownText]="markdownText" />
    `,
    changeDetection: ChangeDetectionStrategy.OnPush
})
class GenerateHTMLFromMarkdownString {
    readonly markdownText: string = `
# H1

## H2

### H3

#### H4

##### H5

###### H6

**bold text**&#32;&#32;
__bold text__

_italic text_&#32;&#32;
*italic text*

***bold and italic text***&#32;&#32;
___bold and italic text___

> blockquote

1. First item
2. Second item
3. Third item

-   First item
-   Second item
-   Third item

\`inline code\`

First line&#32;&#32;
Second line

---

[title](https://www.koobiq.io)

![](https://koobiq.io/assets/images/koobiq-illustration-wip.png)

![With caption text](https://koobiq.io/assets/images/koobiq-illustration-wip.png)
_Image caption text_

| Syntax    | Description | Left-aligned | Center-aligned | Right-aligned |
| --------- | ----------- | :----------- | :------------: | ------------: |
| Header    | Title       | git status   |   git status   |    git status |
| Paragraph | Text        | git diff     |    git diff    |      git diff |

\`\`\`bash
npm install jquery
\`\`\`
    `;
}

@Component({
    standalone: true,
    imports: [KbqMarkdownModule],
    selector: 'generate-html-from-markdown-inline-template',
    // prettier-ignore
    template: `
<kbq-markdown ngPreserveWhitespaces>
# H1

## H2

### H3

#### H4

##### H5

###### H6

**bold text**&#32;&#32;
__bold text__

_italic text_&#32;&#32;
*italic text*

***bold and italic text***&#32;&#32;
___bold and italic text___

> blockquote

1. First item
2. Second item
3. Third item

-   First item
-   Second item
-   Third item

\`inline code\`

First line&#32;&#32;
Second line

---

[title](https://www.koobiq.io)

![](https://koobiq.io/assets/images/koobiq-illustration-wip.png)

![With caption text](https://koobiq.io/assets/images/koobiq-illustration-wip.png)
_Image caption text_

| Syntax    | Description | Left-aligned | Center-aligned | Right-aligned |
| --------- | ----------- | :----------- | :------------: | ------------: |
| Header    | Title       | git status   |   git status   |    git status |
| Paragraph | Text        | git diff     |    git diff    |      git diff |

\`\`\`bash
npm install jquery
\`\`\`
</kbq-markdown>
    `,
    changeDetection: ChangeDetectionStrategy.OnPush
})
class GenerateHTMLFromMarkdownInlineTemplate {}

describe(KbqMarkdown.name, () => {
    it('should generate html from markdown string', async () => {
        const { debugElement } = await createComponent(GenerateHTMLFromMarkdownString);
        expect(getMarkdownDebugElement(debugElement)).toMatchSnapshot();
    });

    it('should generate html from markdown inline template', async () => {
        const { debugElement } = await createComponent(GenerateHTMLFromMarkdownInlineTemplate);
        expect(getMarkdownDebugElement(debugElement)).toMatchSnapshot();
    });
});
