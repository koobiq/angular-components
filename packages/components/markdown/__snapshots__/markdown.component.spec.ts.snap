// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KbqMarkdown should generate html from markdown inline template 1`] = `
DebugElement {
  "nativeNode": <kbq-markdown
    class="kbq-markdown"
  >
    <pre
      class="markdown-input"
    >
      
# H1

## H2

### H3

#### H4

##### H5

###### H6

**bold text**  
__bold text__

_italic text_  
*italic text*

***bold and italic text***  
___bold and italic text___

&gt; blockquote

1. First item
2. Second item
3. Third item

-   First item
-   Second item
-   Third item

\`inline code\`

First line  
Second line

---

[title](https://www.koobiq.io)

![](https://koobiq.io/assets/images/koobiq-illustration-wip.png)

![With caption text](https://koobiq.io/assets/images/koobiq-illustration-wip.png)
_Image caption text_

| Syntax    | Description | Left-aligned | Center-aligned | Right-aligned |
| --------- | ----------- | :----------- | :------------: | ------------: |
| Header    | Title       | git status   |   git status   |    git status |
| Paragraph | Text        | git diff     |    git diff    |      git diff |

\`\`\`bash
npm install jquery
\`\`\`

    </pre>
    <div
      class="markdown-output"
    >
      <h1
        class="kbq-markdown__h1"
      >
        H1
      </h1>
      

      <h2
        class="kbq-markdown__h2"
      >
        H2
      </h2>
      

      <h3
        class="kbq-markdown__h3"
      >
        H3
      </h3>
      

      <h4
        class="kbq-markdown__h4"
      >
        H4
      </h4>
      

      <h5
        class="kbq-markdown__h5"
      >
        H5
      </h5>
      

      <h6
        class="kbq-markdown__h6"
      >
        H6
      </h6>
      

      <p
        class="kbq-markdown__p"
      >
        <strong>
          bold text
        </strong>
        <br />
        <strong>
          bold text
        </strong>
      </p>
      

      <p
        class="kbq-markdown__p"
      >
        <em>
          italic text
        </em>
        <br />
        <em>
          italic text
        </em>
      </p>
      

      <p
        class="kbq-markdown__p"
      >
        <em>
          <strong>
            bold and italic text
          </strong>
        </em>
        <br />
        <em>
          <strong>
            bold and italic text
          </strong>
        </em>
      </p>
      

      <blockquote
        class="kbq-markdown__blockquote"
      >
        

        <p
          class="kbq-markdown__p"
        >
          blockquote
        </p>
        

      </blockquote>
      

      <ol
        class="kbq-markdown__ol"
      >
        

        <li
          class="kbq-markdown__li"
        >
          First item
        </li>
        

        <li
          class="kbq-markdown__li"
        >
          Second item
        </li>
        

        <li
          class="kbq-markdown__li"
        >
          Third item
        </li>
        

      </ol>
      

      <ul
        class="kbq-markdown__ul"
      >
        

        <li
          class="kbq-markdown__li"
        >
          First item
        </li>
        

        <li
          class="kbq-markdown__li"
        >
          Second item
        </li>
        

        <li
          class="kbq-markdown__li"
        >
          Third item
        </li>
        

      </ul>
      

      <p
        class="kbq-markdown__p"
      >
        <code
          class="kbq-markdown__code"
        >
          inline code
        </code>
      </p>
      

      <p
        class="kbq-markdown__p"
      >
        First line
        <br />
        Second line
      </p>
      

      <hr
        class="kbq-markdown__hr"
      />
      

      <p
        class="kbq-markdown__p"
      >
        <a
          class="kbq-markdown__a"
          href="https://www.koobiq.io"
        >
          title
        </a>
      </p>
      

      <p
        class="kbq-markdown__p"
      >
        <img
          alt=""
          class="kbq-markdown__img"
          src="https://koobiq.io/assets/images/koobiq-illustration-wip.png"
        />
      </p>
      

      <p
        class="kbq-markdown__p"
      >
        <img
          alt="With caption text"
          class="kbq-markdown__img"
          src="https://koobiq.io/assets/images/koobiq-illustration-wip.png"
        />
        

        <em>
          Image caption text
        </em>
      </p>
      

      <table
        class="kbq-markdown__table"
      >
        

        <thead
          class="kbq-markdown__thead"
        >
          

          <tr
            class="kbq-markdown__tr"
          >
            

            <th
              class="kbq-markdown__th"
            >
              Syntax
            </th>
            

            <th
              class="kbq-markdown__th"
            >
              Description
            </th>
            

            <th
              align="left"
            >
              Left-aligned
            </th>
            

            <th
              align="center"
            >
              Center-aligned
            </th>
            

            <th
              align="right"
            >
              Right-aligned
            </th>
            

          </tr>
          

        </thead>
        

        <tbody
          class="kbq-markdown__tbody"
        >
          <tr
            class="kbq-markdown__tr"
          >
            

            <td
              class="kbq-markdown__td"
            >
              Header
            </td>
            

            <td
              class="kbq-markdown__td"
            >
              Title
            </td>
            

            <td
              align="left"
              class="kbq-markdown__td"
            >
              git status
            </td>
            

            <td
              align="center"
              class="kbq-markdown__td"
            >
              git status
            </td>
            

            <td
              align="right"
              class="kbq-markdown__td"
            >
              git status
            </td>
            

          </tr>
          

          <tr
            class="kbq-markdown__tr"
          >
            

            <td
              class="kbq-markdown__td"
            >
              Paragraph
            </td>
            

            <td
              class="kbq-markdown__td"
            >
              Text
            </td>
            

            <td
              align="left"
              class="kbq-markdown__td"
            >
              git diff
            </td>
            

            <td
              align="center"
              class="kbq-markdown__td"
            >
              git diff
            </td>
            

            <td
              align="right"
              class="kbq-markdown__td"
            >
              git diff
            </td>
            

          </tr>
          

        </tbody>
      </table>
      

      <pre
        class="kbq-markdown__pre"
      >
        <code
          class="kbq-markdown__code"
        >
          npm install jquery

        </code>
      </pre>
      

    </div>
  </kbq-markdown>,
}
`;

exports[`KbqMarkdown should generate html from markdown string 1`] = `
DebugElement {
  "nativeNode": <kbq-markdown
    class="kbq-markdown"
  >
    <pre
      class="markdown-input"
    />
    <div
      class="markdown-output"
    >
      <h1
        class="kbq-markdown__h1"
      >
        H1
      </h1>
      

      <h2
        class="kbq-markdown__h2"
      >
        H2
      </h2>
      

      <h3
        class="kbq-markdown__h3"
      >
        H3
      </h3>
      

      <h4
        class="kbq-markdown__h4"
      >
        H4
      </h4>
      

      <h5
        class="kbq-markdown__h5"
      >
        H5
      </h5>
      

      <h6
        class="kbq-markdown__h6"
      >
        H6
      </h6>
      

      <p
        class="kbq-markdown__p"
      >
        <strong>
          bold text
        </strong>
          

        <strong>
          bold text
        </strong>
      </p>
      

      <p
        class="kbq-markdown__p"
      >
        <em>
          italic text
        </em>
          

        <em>
          italic text
        </em>
      </p>
      

      <p
        class="kbq-markdown__p"
      >
        <em>
          <strong>
            bold and italic text
          </strong>
        </em>
          

        <em>
          <strong>
            bold and italic text
          </strong>
        </em>
      </p>
      

      <blockquote
        class="kbq-markdown__blockquote"
      >
        

        <p
          class="kbq-markdown__p"
        >
          blockquote
        </p>
        

      </blockquote>
      

      <ol
        class="kbq-markdown__ol"
      >
        

        <li
          class="kbq-markdown__li"
        >
          First item
        </li>
        

        <li
          class="kbq-markdown__li"
        >
          Second item
        </li>
        

        <li
          class="kbq-markdown__li"
        >
          Third item
        </li>
        

      </ol>
      

      <ul
        class="kbq-markdown__ul"
      >
        

        <li
          class="kbq-markdown__li"
        >
          First item
        </li>
        

        <li
          class="kbq-markdown__li"
        >
          Second item
        </li>
        

        <li
          class="kbq-markdown__li"
        >
          Third item
        </li>
        

      </ul>
      

      <p
        class="kbq-markdown__p"
      >
        <code
          class="kbq-markdown__code"
        >
          inline code
        </code>
      </p>
      

      <p
        class="kbq-markdown__p"
      >
        First line  
Second line
      </p>
      

      <hr
        class="kbq-markdown__hr"
      />
      

      <p
        class="kbq-markdown__p"
      >
        <a
          class="kbq-markdown__a"
          href="https://www.koobiq.io"
        >
          title
        </a>
      </p>
      

      <p
        class="kbq-markdown__p"
      >
        <img
          alt=""
          class="kbq-markdown__img"
          src="https://koobiq.io/assets/images/koobiq-illustration-wip.png"
        />
      </p>
      

      <p
        class="kbq-markdown__p"
      >
        <img
          alt="With caption text"
          class="kbq-markdown__img"
          src="https://koobiq.io/assets/images/koobiq-illustration-wip.png"
        />
        

        <em>
          Image caption text
        </em>
      </p>
      

      <table
        class="kbq-markdown__table"
      >
        

        <thead
          class="kbq-markdown__thead"
        >
          

          <tr
            class="kbq-markdown__tr"
          >
            

            <th
              class="kbq-markdown__th"
            >
              Syntax
            </th>
            

            <th
              class="kbq-markdown__th"
            >
              Description
            </th>
            

            <th
              align="left"
            >
              Left-aligned
            </th>
            

            <th
              align="center"
            >
              Center-aligned
            </th>
            

            <th
              align="right"
            >
              Right-aligned
            </th>
            

          </tr>
          

        </thead>
        

        <tbody
          class="kbq-markdown__tbody"
        >
          <tr
            class="kbq-markdown__tr"
          >
            

            <td
              class="kbq-markdown__td"
            >
              Header
            </td>
            

            <td
              class="kbq-markdown__td"
            >
              Title
            </td>
            

            <td
              align="left"
              class="kbq-markdown__td"
            >
              git status
            </td>
            

            <td
              align="center"
              class="kbq-markdown__td"
            >
              git status
            </td>
            

            <td
              align="right"
              class="kbq-markdown__td"
            >
              git status
            </td>
            

          </tr>
          

          <tr
            class="kbq-markdown__tr"
          >
            

            <td
              class="kbq-markdown__td"
            >
              Paragraph
            </td>
            

            <td
              class="kbq-markdown__td"
            >
              Text
            </td>
            

            <td
              align="left"
              class="kbq-markdown__td"
            >
              git diff
            </td>
            

            <td
              align="center"
              class="kbq-markdown__td"
            >
              git diff
            </td>
            

            <td
              align="right"
              class="kbq-markdown__td"
            >
              git diff
            </td>
            

          </tr>
          

        </tbody>
      </table>
      

      <pre
        class="kbq-markdown__pre"
      >
        <code
          class="kbq-markdown__code"
        >
          npm install jquery

        </code>
      </pre>
      

    </div>
  </kbq-markdown>,
}
`;
